wagtail/api/v2/filters.py

```diff
@@ -187,14 +187,50 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class TranslationOfFilter(BaseFilterBackend):
+    """
+    Implements the ?translation_of filter which limits the set of pages to translations
+    of a page.
+    """
+    def filter_queryset(self, request, queryset, view):
+        if 'translation_of' in request.GET:
+            try:
+                page_id = int(request.GET['translation_of'])
+                if page_id < 0:
+                    raise ValueError()
+
+                page = view.get_base_queryset().get(id=page_id)
+            except ValueError:
+                if request.GET['translation_of'] == 'root':
+                    page = view.get_root_page()
+                else:
+                    raise BadRequestError("translation_of must be a positive integer")
+            except Page.DoesNotExist:
+                raise BadRequestError("translation_of page doesn't exist")
+
+            _filtered_by_child_of = getattr(queryset, '_filtered_by_child_of', None)
+
+            queryset = queryset.translation_of(page)
+
+            if _filtered_by_child_of:
+                queryset._filtered_by_child_of = _filtered_by_child_of
+
+        return queryset
+
+
 class LocaleFilter(BaseFilterBackend):
     """
     Implements the ?locale filter which limits the set of pages to a
     particular locale.
     """
     def filter_queryset(self, request, queryset, view):
         if 'locale' in request.GET:
+            _filtered_by_child_of = getattr(queryset, '_filtered_by_child_of', None)
```

Comment:
Maybe out of scope for this PR, but worth considering for cleanup... as far as I can see, `_filtered_by_child_of` is only used to prevent people from passing both `child_of` and `descendant_of` parameters. Doing that is a bit pointless, but surely not logically invalid or ambiguous (all other conditions are ANDed together, so these should be too) - so why are we going to such great lengths (and cluttering up unrelated filter code) to disallow it?

Reply:
I agree, I do like the way Wagtail API is good at giving you errors when you do something wrong but it definitely goes too far in a couple of places! I'll clean this up in another PR!